packages/@angular/cli/bin/ng

```diff
@@ -98,8 +98,11 @@ if (packageJsonProjectPath && fs.existsSync(packageJsonProjectPath)) {
   const hasDep = !!deps['@angular/cli'];
   const hasOldDevDep = !!devDeps['angular-cli'];
   const hasDevDep = !!devDeps['@angular/cli'];
+  // prioritize pulling from project configuration if available otherwise fallback to global
+  const baseConfiguration = CliConfig.fromProject() || CliConfig.fromGlobal();
+  const hasDependencyWarning = baseConfiguration.get('warnings.cliDependency');
 
-  if (hasOldDep || hasOldDevDep || !(hasDevDep || hasDep)) {
+  if (hasDependencyWarning && ( hasOldDep || hasOldDevDep || !(hasDevDep || hasDep) )) {
     const warnings = [
       'Unable to find "@angular/cli" in devDependencies.',
```

Comment:
Should the message be updated to reflect that this actually checks both `dependencies` and `devDepedencies`?

Reply:
Looks to be the case. Should I make the change?
'Unable to find "@angular/cli" in devDependencies.' => 'Unable to find "@angular/cli" in devDependencies or dependencies.'

Reply:
might as well while this section is being updated

Reply:
Alrighty updated 👍